### ISSUE TITLE
Savefiledilaog.ShowDialog() is failing on Windows Server Core RS4. It requires COMDLg32.dll.

### ISSUE THREAD

**Issue by GaneshTambat** [Microsoft/dotnet-framework-early-access#23]
# Issue Title:
Savefiledilaog.ShowDialog() is failing on Windows Server Core RS4. It requires COMDLg32.dll.

Provide a descriptive title:
Savefiledilaog.ShowDialog() is failing on Windows Server Core RS4. It requires COMDLg32.dll.

# Version info 
Window sServer Core RS4 (Build #17101)

Provide the .NET Framework Early Access build # and your Windows version (output of `winver`).
Window sServer Core RS4 (Build #17101)

# General

Savefiledilaog.ShowDialog() is failing. It requires COMDLg32.dll. 
 

If I try to register DLL it throws error :-

# Early Access ID

Some users have been given a .NET Framework Early Access ID. Please provide the ID if you have one.

**Comment by merriemcgaw** [comment]
Hi @GaneshTambat,

Thank you for reporting the issue. You are correct that SaveFileDialog.ShowDialog() requires comdlg32.dll on the system. Windows Server Core does not support the full UI stack in Windows, including not supporting comdlg32.dll (Common Controls). So, this error would be fully expected on ServerCore. I wouldn't expect this to work on earlier versions of ServerCore either.

**Comment by GaneshTambat** [comment]
Right. But is there any chance this will be supported in future RS4 or RS5 builds ?

**Comment by GaneshTambat** [comment]
I would also like to know more about this forum. Can we consider this official response from Microsoft. I am sorry but am quite new to this hence just want to re confirm.

**Comment by kpreisser** [comment]
Hi, you can try to use the WinForms FileDialog which allows to use the legacy (non-Vista) dialog that should work on Windows Server Core, by disabling the `AutoUpgradeEnabled` property.

For example, you can use an extension method like this:
```c#
    using System.Runtime.InteropServices;
    using System.Windows.Forms;

    internal static class FileDialogExtension
    {
        public static DialogResult ShowDialogEx(this FileDialog fileDialog, IWin32Window owner = null)
        {
            try {
                if (owner == null)
                    return fileDialog.ShowDialog();
                else
                    return fileDialog.ShowDialog(owner);
            }
            catch (COMException) when (fileDialog.AutoUpgradeEnabled) {
                // This can happen on Windows Server Core installations which don't have the
                // Vista Dialog. In this case, we fall back to the legacy file dialog which
                // is supported on these platforms.
                fileDialog.AutoUpgradeEnabled = false;
                try {
                    if (owner == null)
                        return fileDialog.ShowDialog();
                    else
                        return fileDialog.ShowDialog(owner);
                }
                finally {
                    fileDialog.AutoUpgradeEnabled = true;
                }                
            }
        }
    }
```

and then call `ShowDialogEx()` instead of `ShowDialog()`.
